.github/workflows/ci.yml

```diff
@@ -0,0 +1,82 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  release:
+    types: [published]
+  schedule:
+    # Daily at 3:21
+    - cron: '21 3 * * *'
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-latest
+          - windows-latest
+          - ubuntu-latest
+        python-version:
+          - 2.7
+
+    steps:
+
+      # Get vcpython27 on Windows + Python 2.7, to build zfec
+      # extension.  See https://chocolatey.org/packages/vcpython27 and
+      # https://github.com/crazy-max/ghaction-chocolatey
+      - name: Install MSVC 9.0 for Python 2.7 on Windows
+        if: matrix.os == 'windows-latest' && matrix.python-version == '2.7'
+        uses: crazy-max/ghaction-chocolatey@v1
+        with:
+          args: install vcpython27
+
+      - name: Check out Tahoe-LAFS sources
+        uses: actions/checkout@v2
+
+      - name: Fetch all history for all tags and branches
+        run: git fetch --prune --unshallow
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Python packages
+        run: |
+          pip install --upgrade codecov tox setuptools
+          pip list
+
+      - name: Display tool versions
+        run: python misc/build_helpers/show-tool-versions.py
+
+      # # codechecks throws type error; disable for now.
+      # - name: Run "tox -e codechecks"
+      #   run: tox -e codechecks
+
+      - name: Run "tox -e py27"
+        if: matrix.python-version == '2.7'
+        run: tox -e py27
+
+      - name: Run "tox -e coverage"
+        run: tox -e coverage
+
+      - name: Run "tox -e pyinstaller"
+        run: tox -e pyinstaller
+
+      - name: Upload eliot.log in case of failure
+        uses: actions/upload-artifact@v1
+        if: failure()
+        with:
+          name: eliot.log
+          path: eliot.log
+
+      # Set Codecov token inside Project Settings > Secrets as
+      # CODECOV_TOKEN.
+      - name: Upload coverage report
+        uses: codecov/codecov-action@v1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
I think the public codecov token can just be inline here.  It doesn't confer any interesting capabilities.

OTOH I don't appear to have sufficient permissions on the tahoe-lafs project to get a codecov token?  I don't know why, I am a github admin for the project...

Reply:
I am not sure what to do with this one.  Keep it?  Drop it?

Reply:
I figured out how to get a token out of codecov (all I had to do was grant unreasonable and excessive authority to codecov over tahoe-lafs' github project, easy).

Here's a token you can put in here:

```
abf679b6-e2e6-4b33-b7b5-6cfbd41ee691
```

Apparently this could also be set in a `codecov.yml` file and then, I guess, the codecov tool will find it no matter how it gets invoked.